File: C++/3005.cpp
#include <vector>
#include <unordered_map>

class Solution {
public:
    int maxFrequencyElements(std::vector<int>& nums) {
        std::unordered_map<int, int> nums_count;
        int current_max = 0;
        int ans = 0;

        for (int num : nums) {
            nums_count[num]++;
        }
        for (const auto& value : nums_count) {
            if (value.second > current_max) {
                current_max = value.second;
                ans = value.second;
            }
            else if(value.second == current_max) {
                ans += value.second;
            }
        }
        return ans;
    }
};
// Topic: Hash Table
// Time Complexity: O(n)
// Space Complexity: O(n)

File: Python/3005.py
from collections import Counter

class Solution:
    def maxFrequencyElements(self, nums: list[int]) -> int:
        nums_count = Counter(nums)
        ans = 0
        current_max = 0

        for value in nums_count.values():
            if value > current_max:
                current_max = value
                ans = value
            elif value == current_max:
                ans += value

        return ans

# Topic: Hash Table
# Time Complexity: O(n)
# Space Complexity: O(n)